deluge/core/alertmanager.py

```diff
@@ -55,17 +56,42 @@ def __init__(self):
         # handlers is a dictionary of lists {"alert_type": [handler1,h2,..]}
         self.handlers = {}
         self.delayed_calls = []
+        self._event = threading.Event()
 
     def update(self):
         self.delayed_calls = [dc for dc in self.delayed_calls if dc.active()]
-        self.handle_alerts()
+
+    def start(self):
+        thread = threading.Thread(
+            target=self.wait_for_alert_in_thread, name='alert-poller'
+        )
+        thread.daemon = True
+        thread.start()
+        self._event.set()
 
     def stop(self):
         for delayed_call in self.delayed_calls:
             if delayed_call.active():
                 delayed_call.cancel()
         self.delayed_calls = []
 
+    def pause(self):
+        self._event.clear()
+
+    def resume(self):
+        self._event.set()
+
+    def wait_for_alert_in_thread(self):
+        while self._component_state not in ('Stopping', 'Stopped'):
+            if self.session.wait_for_alert(1000) is None:
```

Comment:
@aresch I saw you've used `set_alert_notify` in spritzle... better than timer perhaps?

Reply:
Yea, I would suggest using it but I'm not sure if it's in a released libtorrent. I think it would only be available in 1.2 with this commit: https://github.com/arvidn/libtorrent/commit/c2ea38fdfe4cc8f5bd9a9525f6d179ebcf4ab4d6#diff-87803408b3e578d01ff14ecd078cfb4c

We probably don't want to raise the minimum libtorrent dependency to 1.2 quite yet.

Reply:
Hmm well it is in 1.1 so that binding should have be exposed there...

Reply:
@aresch The `wait_for_alert` is now `1s` whereas the AlertManager was polling at `0.3s`, do you think this is perhaps too long?

Reply:
I don't think it's a problem since the only thing done in update() is the clean-up of delayed calls. I was actually going to propose removing that entirely but it was out of scope for this PR. I'll mention that idea now though.

We could wrap all handler functions with another function that removes the delayed call from the list on completion which would remove the need to run a clean-up on the list. We'd probably want to change delayed_calls to a dictionary to make this work efficiently.

```python
def run_handler(self, handler, alert):
    handler_func(alert)
    del self.delayed_calls[(handler, alert)]
```

In handle_alerts() we'd change the creation of the delayed call as such:
```python
for handler in self.handlers[alert_type]:
  self.delayed_calls[(handler, alert)] = reactor.callLater(0, self.run_handler, handler, alert)
```

Anyways, if we think this is a good idea we should do it in a separate PR.

Reply:
> Hmm well it is in 1.1 so that binding should have be exposed there...

I don't see it in the RC_1_1 branch, I think it's only in 1.2.

Reply:
I mean it is in C++ code for [RC_1_1](https://github.com/arvidn/libtorrent/blob/RC_1_1/include/libtorrent/session_handle.hpp#L960) so the binding should have been fixed there

Reply:
Oh, yea, we'd need to add it to the bindings. I think that for now, this approach is fine and there isn't much advantage to switching to use set_alert_notify.